app/models/activity.rb

```diff
@@ -60,19 +60,25 @@ def schedule=(ice_cube_schedule)
     super(ice_cube_schedule.to_yaml)
   end
 
-  def to_time_ranges
-    schedule.occurrences_between(plan.start_date.beginning_of_day, plan.end_date.end_of_day).map do |o|
-      TimeRange.new(
-        start_time: o.start_time,
-        end_time: o.end_time,
-        value: o.duration / 60, # Duration in minutes.
-        user_id: plan.user_id
-      )
+  def to_time_ranges # rubocop:disable Metrics/AbcSize
+    Rails.cache.fetch(time_ranges_cache_key, expires_in: 1.week) do
+      schedule.occurrences_between(plan.start_date.beginning_of_day, plan.end_date.end_of_day).map do |o|
+        TimeRange.new(
+          start_time: o.start_time,
+          end_time: o.end_time,
+          value: o.duration / 60, # Duration in minutes.
+          user_id: plan.user_id
+        )
+      end
     end
   end
 
   private
 
+  def time_ranges_cache_key
+    "Activity#to_time_ranges##{id}"
```

Comment:
Need to bust cache on change:

```suggestion
    "Activity#to_time_ranges##{id}##{updated_at.to_f}""
```

Reply:
good point